tracer/test/Datadog.Trace.Tests/Configuration/DynamicConfigurationTests.cs

```diff
@@ -110,28 +110,56 @@ public void EnableTracing()
         [Fact]
         public void SetSamplingRules()
         {
-            var tracerSettings = new TracerSettings();
+            // start with local sampling rules only
+            var localSamplingRulesConfig = new[]
+            {
+                new { sample_rate = 0.5,  service = "Service3", resource = "Resource3", },
+            };
+
+            var localSamplingRulesJson = JsonConvert.SerializeObject(localSamplingRulesConfig);
+            var configValues = new Dictionary<string, string> { { "DD_TRACE_SAMPLING_RULES", localSamplingRulesJson } };
+
+            IConfigurationSource configSource = new DictionaryConfigurationSource(configValues);
+            var tracerSettings = new TracerSettings(configSource);
             TracerManager.ReplaceGlobalManager(new ImmutableTracerSettings(tracerSettings), TracerManagerFactory.Instance);
```

Comment:
Also unrelated, but this test fails locally a lot for me, because we're still reaching out to a _real_ agent, which means I get _real_ remote sampling rules 😅 One way around this would be to call `TracerManagerFactory.CreateTracerManager(..)` and provide mock objects... Obviously not required to fix here, just generally thinking about it.. I wonder if we should stop using all the static instances in these tests in fact 🤔